homeassistant/components/crownstone/light.py

```diff
@@ -0,0 +1,173 @@
+"""Support for Crownstone devices."""
+import logging
+from typing import Any, Dict, Optional
+
+from crownstone_cloud.const import DIMMING_ABILITY
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    SUPPORT_BRIGHTNESS,
+    LightEntity,
+)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import CROWNSTONE_EXCLUDE, CROWNSTONE_TYPES, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up crownstones from a config entry."""
+    crownstone_hub = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for crownstone in crownstone_hub.sphere.crownstones:
+        # some don't support light features
+        if crownstone.type not in CROWNSTONE_EXCLUDE:
+            entities.append(
+                Crownstone(crownstone, crownstone_hub.uart_manager.uart_instance)
+            )
+
+    async_add_entities(entities, True)
+
+
+def crownstone_state_to_hass(value: float):
+    """Crownstone 0..1 to hass 0..255."""
+    return value * 255
+
+
+def hass_to_crownstone_state(value: float):
+    """Hass 0..255 to Crownstone 0..1."""
+    return value / 255
+
+
+class Crownstone(LightEntity):
+    """
+    Representation of a crownstone.
+
+    Light platform is used to support dimming.
+    """
+
+    def __init__(self, crownstone, uart):
+        """Initialize the crownstone."""
+        self.crownstone = crownstone
+        self.uart = uart
+
+    @property
+    def name(self) -> str:
+        """Return the name of this crownstone."""
+        return self.crownstone.name
+
+    @property
+    def icon(self) -> Optional[str]:
+        """Return the icon."""
+        return "mdi:power-socket-de"
+
+    @property
+    def type(self) -> str:
+        """Return the crownstone type."""
+        return CROWNSTONE_TYPES[self.crownstone.type]
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID."""
+        return self.crownstone.unique_id
+
+    @property
+    def cloud_id(self) -> str:
+        """Return the cloud id of this crownstone."""
+        return self.crownstone.cloud_id
+
+    @property
+    def should_dim(self) -> bool:
+        """Return if this crownstone is able to dim."""
+        return self.crownstone.abilities.get(DIMMING_ABILITY).is_enabled
```

Comment:
The change in abilities affects the `supported_features` dynamically.
This is not allowed.

Reply:
What is the reason this isn't allowed?
Changing abilities on the fly is an important part of Crownstones.
Not supporting brightness by default will make dimming impossible, and leaving it on 1 by default will cause misunderstandings of whether it is enabled or not, e.g. people trying to set a brightness but the Crownstone just turns on at 100% brightness.
I have the Crownstone integration in use on my Raspberry Pi for a couple weeks as custom component, and i have not experienced any problems with dynamically changing `supported_features`.

What is your recommendation of approaching this problem differently?

Reply:
Determine the abilities upfront and keep it static for the lifetime.